diff --git a/node_modules/@jup-ag/core/dist/core.esm.js b/node_modules/@jup-ag/core/dist/core.esm.js
index 0dd7743..ec94ed9 100644
--- a/node_modules/@jup-ag/core/dist/core.esm.js
+++ b/node_modules/@jup-ag/core/dist/core.esm.js
@@ -1363,17 +1363,17 @@ function raydiumAmmToRaydiumSwap(raydiumAmm, userSourceTokenAccountAddress, user
 }
 
 function marketIntoSerumSwap(market, openOrdersAddress, orderPayerTokenAccountAddress, coinWallet, pcWallet, user) {
-  const vaultSigner = createProgramAddressSync([market.address.toBuffer(), market.decoded.vaultSignerNonce.toArrayLike(Buffer, 'le', 8)], market.programId);
+  const vaultSigner = createProgramAddressSync([market.address.toBuffer(), market._decoded.vaultSignerNonce.toArrayLike(Buffer, 'le', 8)], market.programId);
   return {
     market: {
       market: market.address,
       openOrders: openOrdersAddress,
-      requestQueue: market.decoded.requestQueue,
-      eventQueue: market.decoded.eventQueue,
+      requestQueue: market._decoded.requestQueue,
+      eventQueue: market._decoded.eventQueue,
       bids: market.bidsAddress,
       asks: market.asksAddress,
-      coinVault: market.decoded.baseVault,
-      pcVault: market.decoded.quoteVault,
+      coinVault: market._decoded.baseVault,
+      pcVault: market._decoded.quoteVault,
       vaultSigner
     },
     authority: user,
